### ISSUE TITLE
Bond causes SIGBUS whenever a double is serialized from Java to c++

### ISSUE THREAD

**Issue by ajpeacock0** [Microsoft/bond#511]
Hi Folks,

I am working on a CloudDataSet Wrapper to serialize Java primitives to C++. All data types work except for doubles. When a CloudDataSet is given to `cpp\inc\bond\core\bond.h` containing any `double` or `double` array then a SIGBUS is hit inside `cpp\inc\bond\stream\output_buffer.h`'s write function

```
void Write(const T& value)
    {
        //
        // Performance tweak: for small types don't go into generic Write(),
        // which results in call to memcpy() and additional memory access.
        // The direct copy of the value (which is likely on register already)
        // is faster.
        //
        if (sizeof(T) + _rangeSize + _rangeOffset <= _bufferSize)
        {
            *reinterpret_cast<T*>(_rangePtr + _rangeSize) = value; <--------HERE---------
            _rangeSize += sizeof(T);
        }
        else
        {
            Write(&value, sizeof(value));
        }
    }
```

This appears to be a bug inside the bond serialize code.

The local values are
```
this = {bond::OutputMemoryStream<std::allocator<char> > *const | 0xbe9953c0} 0xbe9953c0
 _allocator = {std::allocator<char>} 
 _buffer = {boost::shared_ptr<char []>} 
 _bufferSize = {uint32_t} 4096
 _rangeSize = {uint32_t} 234
 _rangeOffset = {uint32_t} 0
 _minChainningSize = {uint32_t} 32
 _maxChainLength = {uint32_t} 4294967295
 _rangePtr = {char * | 0xb2b60814} "-\x12\n\x04\fC"
 _blobs = {std::vector<bond::blob>} std::vector of length 0, capacity 0
value = {const double &const} 1.2344999999999999
```

Attached is the callstack and last frame hit in Android Studio. Let me know if this is a known issue or not.

Thanks,
Andrew

![android_studio](https://user-images.githubusercontent.com/7133924/27556474-9d0f41dc-5a6a-11e7-80a6-509fbd51c52e.png)

**Comment by chwarr** [comment]
This looks like https://github.com/Microsoft/bond/issues/305, which I attempted to fix in [version 5.3.0](https://github.com/Microsoft/bond/blob/master/CHANGELOG.md#530-2017-04-12) (commit https://github.com/Microsoft/bond/commit/04ad85fe75a72c70068fce9e07e0cb07379c8b0a). The screenshot and code snippet appear to be from an earlier version than 5.3.0. Can you try with the current version, 5.3.1, and let us know if you still have the problem?

**Comment by ajpeacock0** [comment]
Build 5.3.1 and that fixed the issue! Thanks.
